### PR TITLE
Handle long query texts better.

### PR DESCRIPTION
**Pull request by isoos**
Screenshot 2020-07-27 at 15 19 08

### CODE REVIEW

File: app/lib/search/search_client.dart [hunk 1]
```diff
@@ -62,6 +62,13 @@ class SearchClient {
       return result;
     }
 
+    // Block search on unreasonably long search queries (when the free-form
+    // text part is longer than one would enter via the search input field).
+    final queryLength = query?.parsedQuery?.text?.length ?? 0;
+    if (queryLength > 256) {
```
Comment (sigurdm) [inline app/lib/search/search_client.dart:68]: Can we extract and name this magic constant?

File: app/test/frontend/golden/pkg_index_page.html [hunk 2]
```diff
@@ -179,6 +179,7 @@ <h3 class="foldable-button">Dart
     packages
 
 
+
```
Comment (sigurdm) [inline app/test/frontend/golden/pkg_index_page.html:182]: Is this extra newline intentional?

Comment (isoos) [reply app/test/frontend/golden/pkg_index_page.html:182]: That is coming from the the mustache template. We could optimize for nicer outputs but then the source templates would be less readable.

Comment (isoos) [reply app/test/frontend/golden/pkg_index_page.html:182]: And yes, it is expected from the template change (but not intentional).

File: app/lib/frontend/templates/views/shared/listing_info.mustache [hunk 3]
```diff
@@ -15,6 +15,10 @@
     {{#has_owned_by}}
     owned by <code>{{ owned_by }}</code>
     {{/has_owned_by}}
+
```
Comment (sigurdm) [inline app/lib/frontend/templates/views/shared/listing_info.mustache:18]: Could you not move the newline into the conditional>?

**Pull request merged**